webui/react/src/hooks/useCreateExperimentModal.tsx

```diff
@@ -0,0 +1,350 @@
+    searcher: {
+      max_length: experimentConfig.searcher.max_length,
+      metric: experimentConfig.searcher.metric,
+      name: 'single',
+      smaller_is_better: experimentConfig.searcher.smaller_is_better,
+      source_trial_id: trialId,
+    },
+  };
+};
+
+const MonacoEditor = React.lazy(() => import('react-monaco-editor'));
+
+const useCreateExperimentModal = (): ModalHooks => {
+  const modalRef = useRef<ReturnType<ModalFunc>>();
+  const formRef = useRef<FormInstance>(null);
+  const [ modalState, setModalState ] = useState<ModalState>({
+    config: {},
+    configString: '',
+    isAdvancedMode: false,
+    type: CreateExperimentType.Fork,
+    visible: false,
+  });
+  const prevIsAdvancedMode = usePrevious(modalState.isAdvancedMode, false);
+
+  const showModal = useCallback(({ experiment, trial, type }: ShowProps) => {
+    const isFork = type === CreateExperimentType.Fork;
+
+    let config = clone(experiment.configRaw);
+    if (!isFork && trial) {
+      config = trialContinueConfig(config, trial.hyperparameters, trial.id);
+      config.description = `Continuation of trial ${trial.id}, experiment ${experiment.id}` +
+        (config.description ? ` (${config.description})` : '');
+    } else if (!isFork) {
+      if (config.description) config.description = `Fork of ${config.description}`;
+    }
+    upgradeConfig(config);
+
+    setModalState({
+      config,
+      configString: yaml.dump(config),
+      experiment,
+      isAdvancedMode: false,
+      trial,
+      type,
+      visible: true,
+    });
+  }, []);
+
+  const closeModal = useCallback(() => {
+    if (!modalRef.current) return;
+    modalRef.current.destroy();
+    modalRef.current = undefined;
+  }, []);
+
+  const getConfigFromForm = useCallback((config: RawJson) => {
+    if (!formRef.current) return yaml.dump(config);
+
+    const formValues = formRef.current.getFieldsValue();
+    const newConfig = clone(config);
+
+    if (formValues.name) {
+      newConfig.name = formValues.name;
+    }
+    if (formValues.maxLength) {
+      const maxLengthType = getMaxLengthType(newConfig);
+      newConfig.searcher.max_length = { [maxLengthType]: parseInt(formValues.maxLength) };
+    }
+
+    return yaml.dump(newConfig);
+  }, []);
+
+  const submitExperiment = useCallback(async (newConfig: string) => {
+    const isContinueTrial = modalState.type === CreateExperimentType.ContinueTrial;
+    if (!modalState.experiment || (isContinueTrial && !modalState.trial)) return;
+
+    try {
+      const { id: newExperimentId } = await createExperiment({
+        experimentConfig: newConfig,
+        parentId: modalState.experiment.id,
+      });
+
+      closeModal();
+
+      // Route to reload path to forcibly remount experiment page.
+      const newPath = paths.experimentDetails(newExperimentId);
+      routeToReactUrl(paths.reload(newPath));
+    } catch (e) {
+      let errorMessage = 'Unable to continue trial with the provided config.';
```

Comment:
would this be triggered for fork modals as well?

Reply:
ah yeah would have to distinguish the two, nice catch!